### PR TITLE
catch invite error and retry

### LINKED ISSUE #83: members:set is broken
```
maciek@mothra:~$ heroku members:set --role admin aquint@heroku.com --org dod
Inviting aquint@heroku.com to dod as admin... !
 ▸    aquint@heroku.com user cannot be invited because they are already in dod team
```
It looks like there was an API change that means you can't "upsert" roles anymore, which seems to be the behavior that `members:set` was relying on (since it's just a copy of `members:add`).

### PR DESCRIPTION
**Pull request by RasPhilCo**
Closes #83, see new rules in comments below

### CODE REVIEW

**Comment by jdx** [comment]
Can you add a test?

File: commands/members/add.js [hunk 1]
```diff
@@ -27,17 +27,24 @@ function * run (context, heroku) {
...
-      headers: {
-        Accept: 'application/vnd.heroku+json; version=3.team-invitations'
-      },
-      method: 'PUT',
-      path: `/organizations/${groupName}/invitations`,
-      body: {email, role}
-    }).then(request => {
+  let inviteMemberToTeam = async function (email, role, groupName) {
+    try {
+      yield heroku.request({
```
Comment (jdx) [inline commands/members/add.js]: Should be await if you’re using async

**Review by jdx — CHANGES_REQUESTED** [review]

**Comment by kahlouie** [comment]
@RasPhilCo @jdxcode Shoot, I just double checked. We don't error on the `inviteMemberToTeam` endpoint if the team doesn't use invitations (Heroku Teams), which means it will just work. I can either add that in, or we can brainstorm another solution. I recommend adding that in since we shouldn't be creating team invitations for Enterprise teams.

**Comment by RasPhilCo** [comment]
| command (x-axis)/ org type (y) | orgs:add | orgs:set  | 
|---|---|---|
| team w/ invites feature | PUT /teams/:id/invitations | PATCH or PUT /teams/:id/members |  
| team w/o invites feature|  POST or PUT /teams/:id/members |  PATCH or PUT /teams/:id/members |  
| enterprise team  | POST or PUT /teams/:id/members |  PATCH or PUT /teams/:id/members |

**Comment by RasPhilCo** [comment]
@kahlouie I made a table of api endpoints given org type and command called, does this look right or are we missing a command or missing an endpoint or missing a workflow? (If the table is a more distraction than help let me know) Feel free to edit it (i think you can edit my post)

**Comment by kahlouie** [comment]
@RasPhilCo I edited the table. We should be switching to teams endpoints anyway now. :)

**Comment by RasPhilCo** [comment]
@kahlouie 👍  will update PR

**Comment by RasPhilCo** [comment]
@kahlouie 4 things:

1) I added this issue https://github.com/heroku/heroku-orgs/issues/86 to update all the endpoints. The orgs endpoint test mock is shared across all the command tests and updating it here was confusing the point of this PR.
2) Below I've included the test descriptions. I want to make sure we're covering what we think we are covering.
3) Re Maciek's issue that started this https://github.com/heroku/heroku-orgs/issues/83, this won't fix that, he'll still get the warning that the team member can't be invited. But maybe it's a wont-fix as `set` is the command that should be used. (Tbh, it's still kinda confusing because the `members:set` command still UPSERTS/creates, as Maciek pointed out. And `members:add` is confusing because if you're a team w/o the invite feature, it will work to set the role. Not that the endpoint should change behavior, per say, just high-lighting any place where we may confuse users.)
4) ~I'm not sure i've addressed this point `we shouldn't be creating team invitations for Enterprise teams.`, as the exiting tests seem to imply that's possible. Let me update that.~ Updated

**Comment by RasPhilCo** [comment]
```
  heroku members:add
    without the feature flag team-invite-acceptance
      and group is a team
        ✓ does not warn the user when under the free org limit
        ✓ does not warn the user when over the free org limit
        ✓ does warn the user when at the free org limit
        using --org instead of --team
          ✓ adds the member, but it shows a warning about the usage of -t instead
      and group is an enterprise org
        ✓ adds a member to an org
    with the feature flag team-invite-acceptance for a team
        ✓ does warn the user when free org limit is caused by invites
        ✓ sends an invite when adding a new user to the team

(removed test)
      and group is an enterprise org
        ✓ sends an invite when adding a new user to the team


  heroku members:set
    and group is a team
      ✓ does not warn the user when under the free org limit
      ✓ does not warn the user when over the free org limit
      ✓ does warn the user when at the free org limit
      using --org instead of --team
        ✓ adds the member, but it shows a warning about the usage of -t instead
    and group is an enterprise org
      ✓ adds a member to an org
```

**Review by kahlouie — CHANGES_REQUESTED** [review]
In general, it looks good. I wonder if we should change the set to command to only update existing ones. If we do, we should be using the PATCH '/teams/:team_id/members', which feels better to me.

**Comment by RasPhilCo** [comment]
@kahlouie @jdxcode `members:set` updated to use `PATCH`

**Comment by kahlouie** [comment]
yay! 🎉

**Review by kahlouie — APPROVED** [review]

**Comment by RasPhilCo** [comment]
@kahlouie Cool!

@jdxcode Do you want to take a final look before merging?

**Review by jdx — APPROVED** [review]

**Pull request merged**